scipy/optimize/_trustregion_krylov.py

```diff
@@ -0,0 +1,39 @@
+from ._trustregion import (_minimize_trust_region)
+from ._trlib import (TRLIBQuadraticSubproblem)
+
+__all__ = ['_minimize_trust_krylov']
+
+def _minimize_trust_krylov(fun, x0, args=(), jac=None, hess=None,
+        hessp=None, inexact=True, **trust_region_options):
+    """
+    Minimization of a scalar function of one or more variables using
+    a nearly exact trust-region algorithm that only requires matrix
+    vector products with the hessian matrix.
+
+    Options
+    -------
+    inexact : bool, optional
+        Accuracy to solve subproblems. If True requires less nonlinear
+        iterations, but more vector products.
+
+    .. versionadded:: 1.0.0
+    """
+
+    if jac is None:
+        raise ValueError('Jacobian is required for trust region ',
+                         'exact minimization.')
+    if hess is None and hessp is None:
+        raise ValueError('Either the Hessian or the Hessian-vector product '
+                         'is required for Krylov trust-region minimization')
+    if inexact:
+        return _minimize_trust_region(fun, x0, args=args, jac=jac,
+                hess=hess, hessp=hessp,
+                subproblem=TRLIBQuadraticSubproblem(
+                    tol_rel_i=-2.0, tol_rel_b=-3.0),
```

Comment:
I have included some explanation now in the file. Indeed, these are the convergence tolerances for the krylov subspace iteration relative to the initial gradient, `_i` defines the tolerance for interior convergence while `_b` for boundary convergence.
Settings `-2` and `-3` make the c code select a forcing sequence with quadratic convergence behaviour that turns out to be a good compromise between outer nonlinear iterations and inner krylov subspace iterations (essential cost: 1 matrix vector product / inner iteration).
In general: Tightening these tolerances will lead to an increased convergence speed of outer nonlinear iterations at the cost of additional matrix vector products. One should be cautious not to set the boundary tolerance too high, as the Lanczos procedure might start to accumulate garbage via numerical instability; at least as long as no reorthonomalization is performed.